### PR TITLE
Handle the case of no metadata on master for new connectors

### PR DESCRIPTION
**Pull request by bnchrch**
## What
We had two issues
1. We didnt gracefully handle the null case for tags
2. We didnt gracefully handle when a metadata file was not on master (because the connector was new)

### CODE REVIEW

**Comment by octavia-approvington** [comment]
Myomoto says it looks good <br/> thats a niiice

**Review by octavia-approvington — APPROVED** [review]

File: tools/ci_connector_ops/ci_connector_ops/pipelines/tests/common.py [hunk 1]
```diff
@@ -31,14 +31,20 @@ def github_master_metadata_url(self):
...
 
     @cached_property
-    def master_metadata(self) -> dict:
+    def master_metadata(self) -> Optional[dict]:
         response = requests.get(self.github_master_metadata_url)
-        response.raise_for_status()
+
+        # New connectors will not have a metadata file in master
+        if not response.ok:
+            return None
```
Comment (alafanechere) [inline tools/ci_connector_ops/ci_connector_ops/pipelines/tests/common.py:39]:
```suggestion
        if response.status_code not in [200, 404]:
            response.raise_for_status()
       if response.status_code == 404:
            return None
```

**Review by alafanechere — APPROVED** [review]
Please checkout my suggestion.  LGTM Otherwise. Thanks for the fix.

**Pull request merged**